### PR TITLE
[improvement] In order to achieve more Datax datasource type support

### PR DESCRIPTION
**Pull request by xinxingi**
## Purpose of the pull request
In order to allow more plug-ins to support Datax UI configuration, some codes are disassembled to enhance the functions of this plug-in. Taking Doris as an example, write is written through Http, and some configuration items need to be modified.
https://github.com/alibaba/DataX/blob/master/doriswriter/doc/doriswriter.md

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-task-plugin/dolphinscheduler-task-datax/src/main/java/org/apache/dolphinscheduler/plugin/task/datax/DataxTask.java [hunk 1]
```diff
@@ -243,6 +243,31 @@ private List<ObjectNode> buildDataxJobContentJson() {
...
+        List<ObjectNode> contentList = new ArrayList<>();
+        ObjectNode content = JSONUtils.createObjectNode();
+        content.set("reader", reader);
+        content.set("writer", writer);
+        contentList.add(content);
+
+        return contentList;
+    }
+
+    private ObjectNode buildDataxJobContentJsonDefaultReader(BaseConnectionParam dataSourceCfg) {
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-datax/src/main/java/org/apache/dolphinscheduler/plugin/task/datax/DataxTask.java]: Could you move this from DataXTask? and use a new interface to generate datax file content?

Comment (xinxingi) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-datax/src/main/java/org/apache/dolphinscheduler/plugin/task/datax/DataxTask.java]: ASAP

**Comment by fuchanghai** [comment]
hi @ruanwenjun  @xinxingi I think it is best to write an abstract class to solve this problem, and use a public method for the jdbc type. Non-jdbc uses self-defined methods, the following are some optimizations of this method
 image
image

cc @Radeity @SbloodyS

**Comment by ruanwenjun** [comment]
> hi @ruanwenjun @xinxingi I think it is best to write an abstract class to solve this problem, and use a public method for the jdbc type. Non-jdbc uses self-defined methods, the following are some optimizations of this method image image
> 
> cc @Radeity @SbloodyS

Yes, this kind of logic should be put into a new interface, the DataXTask doesn't care about the content.

**Comment by xinxingi** [comment]
@fuchanghai ACK

**Pull request closed without merging**